From 68974e0717172c13d8ef0b03f4c2a92a12f748df Mon Sep 17 00:00:00 2001
From: "Christopher M. Cantalupo" <christopher.m.cantalupo@intel.com>
Date: Tue, 6 Oct 2020 17:08:13 -0700
Subject: [PATCH 2/3] Increase iteration count for PCG solve

- Increase mg_max_iter from 100 to 500
- Decrease error tolerence to zero to avoid
  leaving loop because of error convergence.

#  Copyright (c) 2015 - 2024, Intel Corporation
#  SPDX-License-Identifier: BSD-3-Clause
#

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
---
 test/amg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/amg.c b/test/amg.c
index 18811ea..842f636 100644
--- a/test/amg.c
+++ b/test/amg.c
@@ -72,7 +72,7 @@ main( hypre_int argc,
    HYPRE_Int           max_levels = 25;
    HYPRE_Int           num_iterations;
    HYPRE_Int           max_iter = 1000;
-   HYPRE_Int           mg_max_iter = 100;
+   HYPRE_Int           mg_max_iter = 500;
    HYPRE_Int 	       cum_num_its=0;
    HYPRE_Int           nodal = 0;
    HYPRE_Real          final_res_norm;
@@ -113,7 +113,7 @@ main( hypre_int argc,
    HYPRE_Int    relax_type = 18;   
    HYPRE_Int    rap2=1;
    HYPRE_Int    keepTranspose = 0;
-   HYPRE_Real   tol = 1.e-8, pc_tol = 0.;
+   HYPRE_Real   tol = 0., pc_tol = 0.;
    HYPRE_Real   atol = 0.0;
 
    HYPRE_Real   wall_time;
-- 
2.26.2

